src/main/java/frc/team2412/robot/subsystems/LauncherSubsystem.java

```diff
@@ -0,0 +1,140 @@
+			Shuffleboard.getTab("Launcher")
+					.addPersistent("is loaded", false)
+					.withSize(1, 1)
+					.withWidget(BuiltInWidgets.kBooleanBox)
+					.getEntry();
+	// Constructor
+	public LauncherSubsystem() {
+
+		// MOTOR INSTANCE VARIBLES
+		launcherTopMotor = new CANSparkFlex(Hardware.LAUNCHER_TOP_MOTOR_ID, MotorType.kBrushless);
+		launcherBottomMotor = new CANSparkFlex(Hardware.LAUNCHER_BOTTOM_MOTOR_ID, MotorType.kBrushless);
+		launcherAngleMotor = new CANSparkFlex(Hardware.LAUNCHER_ANGLE_MOTOR_ID, MotorType.kBrushless);
+		launcherHoodMotor = new CANSparkFlex(Hardware.LAUNCHER_HOOD_MOTOR_ID, MotorType.kBrushless);
+		launcherTopEncoder = launcherTopMotor.getEncoder();
+		launcherBottomEncoder = launcherBottomMotor.getEncoder();
+		launcherAngleEncoder = launcherAngleMotor.getAbsoluteEncoder(Type.kDutyCycle);
+		launcherHoodEncoder =
+				launcherHoodMotor.getAbsoluteEncoder(SparkAbsoluteEncoder.Type.kDutyCycle);
+		launcherAnglePIDController = launcherAngleMotor.getPIDController();
+		launcherHoodPIDController = launcherHoodMotor.getPIDController();
+
+		launcherAnglePIDController.setFeedbackDevice(launcherAngleEncoder);
+		launcherHoodPIDController.setFeedbackDevice(launcherHoodEncoder);
+		configMotors();
+	}
+
+	public void configMotors() {
+		launcherTopMotor.restoreFactoryDefaults();
+		launcherBottomMotor.restoreFactoryDefaults();
+		launcherAngleMotor.restoreFactoryDefaults();
+		launcherHoodMotor.restoreFactoryDefaults();
+
+		// idle mode (wow)
+		launcherTopMotor.setIdleMode(IdleMode.kBrake);
+		launcherBottomMotor.setIdleMode(IdleMode.kBrake);
+		launcherAngleMotor.setIdleMode(IdleMode.kBrake);
+		launcherHoodMotor.setIdleMode(IdleMode.kBrake);
+
+		// current limit
+		launcherTopMotor.setSmartCurrentLimit(20);
+		launcherBottomMotor.setSmartCurrentLimit(20);
+		launcherAngleMotor.setSmartCurrentLimit(20);
+		launcherHoodMotor.setSmartCurrentLimit(20);
+
+		launcherTopMotor.burnFlash();
+		launcherBottomMotor.burnFlash();
+		launcherAngleMotor.burnFlash();
+		launcherHoodMotor.burnFlash();
+	}
+
+	// stop specific motor method
+	public void stopMotor(CANSparkFlex motor) {
+		motor.stopMotor();
+	}
+
+	public void shoot(double speed) {
+		launcherTopMotor.set(speed);
+		launcherBottomMotor.set(-speed);
+	}
+	// returns the degrees of the angle of the launcher
+	public double getAngle() {
+		// get position returns a double in the form of rotations
+		return Units.rotationsToDegrees(launcherAngleEncoder.getPosition());
+	}
+
+	public void setAngle(double angle) {
+		launcherAnglePIDController.setReference(Units.degreesToRotations(angle), ControlType.kPosition);
+	}
+
+	public double getHoodAngle() {
+		return Units.rotationsToDegrees(launcherHoodEncoder.getPosition());
+	}
+
+	public void setHoodAngle(double angle) {
+		launcherHoodPIDController.setReference(Units.degreesToRotations(angle), ControlType.kPosition);
+	}
+
+	@Override
+	public void periodic() {
+		launcherSpeed.setDouble(launcherTopEncoder.getVelocity());
```

Comment:
This is reporting in different units (velocity) than the motor's being set at (raw power). It'd be better to be consistent and use power units here for now.

Reply:
alright, set it to getVelocity for now, also added a comment to remind us to fix it once we establish that.